end_to_end/2-lineage-train-assess-bias-tune-registry-e2e.ipynb

```diff
@@ -0,0 +1,1083 @@
+{
```

Comment:
<p>I got the following error when running this cell:</p><p> </p><pre class="ql-syntax" spellcheck="false">---------------------------------------------------------------------------
ClientError&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;Traceback (most recent call last)
&lt;ipython-input-12-f1cd7a5d394e&gt; in &lt;module&gt;
&nbsp; &nbsp; &nbsp;18&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;source_uri=code_s3_uri,
&nbsp; &nbsp; &nbsp;19&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;artifact_type='Code',
---&gt; 20&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;sagemaker_session=sagemaker_session)
&nbsp; &nbsp; &nbsp;21&nbsp; &nbsp; &nbsp;print(f'Create artifact {code_artifact.artifact_arn}: SUCCESSFUL')


~/anaconda3/lib/python3.7/site-packages/sagemaker/lineage/artifact.py in create(cls, artifact_name, source_uri, source_types, artifact_type, properties, tags, sagemaker_session)
&nbsp; &nbsp; 263&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;properties=properties,
&nbsp; &nbsp; 264&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;tags=tags,
--&gt; 265&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;sagemaker_session=sagemaker_session,
&nbsp; &nbsp; 266&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;)
&nbsp; &nbsp; 267&nbsp;


~/anaconda3/lib/python3.7/site-packages/sagemaker/apiutils/_base_types.py in _construct(cls, boto_method_name, sagemaker_session, **kwargs)
&nbsp; &nbsp; 186&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;sagemaker_session = sagemaker_session or _utils.default_session()
&nbsp; &nbsp; 187&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;instance = cls(sagemaker_session, **kwargs)
--&gt; 188&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;return instance._invoke_api(boto_method_name, kwargs)
&nbsp; &nbsp; 189&nbsp;
&nbsp; &nbsp; 190&nbsp; &nbsp; &nbsp;def _set_tags(self, resource_arn=None, tags=None):


~/anaconda3/lib/python3.7/site-packages/sagemaker/apiutils/_base_types.py in _invoke_api(self, boto_method, boto_method_members)
&nbsp; &nbsp; 222&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;api_kwargs = self.to_boto(api_values)
&nbsp; &nbsp; 223&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;api_method = getattr(self.sagemaker_session.sagemaker_client, boto_method)
--&gt; 224&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;api_boto_response = api_method(**api_kwargs)
&nbsp; &nbsp; 225&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;return self.with_boto(api_boto_response)


~/anaconda3/lib/python3.7/site-packages/botocore/client.py in _api_call(self, *args, **kwargs)
&nbsp; &nbsp; 355&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;"%s() only accepts keyword arguments." % py_operation_name)
&nbsp; &nbsp; 356&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;# The "self" in this scope is referring to the BaseClient.
--&gt; 357&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;return self._make_api_call(operation_name, kwargs)
&nbsp; &nbsp; 358&nbsp;
&nbsp; &nbsp; 359&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;_api_call.__name__ = str(py_operation_name)


~/anaconda3/lib/python3.7/site-packages/botocore/client.py in _make_api_call(self, operation_name, api_params)
&nbsp; &nbsp; 674&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;error_code = parsed_response.get("Error", {}).get("Code")
&nbsp; &nbsp; 675&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;error_class = self.exceptions.from_code(error_code)
--&gt; 676&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;raise error_class(parsed_response, operation_name)
&nbsp; &nbsp; 677&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;else:
&nbsp; &nbsp; 678&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;return parsed_response


ClientError: An error occurred (ValidationException) when calling the CreateArtifact operation: Artifact creation is currently restricted to the SageMaker/Pipeline runtimes.
</pre><p> </p><p> </p><p>On my end, the value of <code>code_s3_uri</code>  is </p><p><code>"s3://sagemaker-us-east-2-688520471316/fraud-detect-demo/training_jobs/sagemaker-xgboost-2021-02-05-00-12-08-738/source/sourcedir.tar.gz"</code>  but it should be a directory as the name <code>sagemaker_submit_directory</code>  suggests;</p><p> </p><p>One more thing: I did run the cell for training the model, so I would expect <code>matching_artifacts</code>  to be non-empty</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/1974/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Reply:
<p>Apologies not getting that error after rerun</p>